src/components/QueryFileUploader/QueryFileUploader.js

```diff
@@ -1,44 +1,168 @@
-import React, { useState } from 'react';
+import React, {
+  useState,
+  useRef,
+  useEffect,
+} from 'react';
 import { FormattedMessage } from 'react-intl';
-import { noop } from 'lodash';
+import PropTypes from 'prop-types';
 
-import { FileUploader } from '@folio/stripes-data-transfer-components';
-import { Layout } from '@folio/stripes-components';
+import {
+  FileUploader,
+  uploadFile,
+  Preloader,
+  createUrl,
+} from '@folio/stripes-data-transfer-components';
+import {
+  Layout,
+  Callout,
+} from '@folio/stripes-components';
+import {
+  useStripes,
+  stripesConnect,
+} from '@folio/stripes-core';
+
+import { generateFileDefinitionBody } from './utils';
 
 import css from './QueryFileUploader.css';
 
-export function QueryFileUploader() {
+function QueryFileUploaderComponent(props) {
+  const { mutator } = props;
+
   const [isDropZoneActive, setDropZoneActive] = useState(false);
+  const [isLoading, setLoading] = useState(false);
+  const calloutRef = useRef(null);
+  const currentFileUploadXhr = useRef(null);
+  const stripes = useStripes();
 
-  const uploaderTitle = isDropZoneActive ?
+  const uploaderTitle = isDropZoneActive ? isLoading ?
+    <Preloader message={<FormattedMessage id="ui-data-export.uploading" />} /> :
```

Comment:
1. The colon and question symbols should be placed at the beginning of the line according to the published rules https://wiki.folio.org/display/FOLIJET/Code+style+rules+not+covered+by+eslint (And even in https://github.com/airbnb/javascript#comparison--nested-ternaries). Do we have special other rules for these cases?
2. Are you sure that these complex ternaries are readable enough? It works for me but looks complicated.

Reply:
1. Good catch. Will do.
2. Replacing it with `if` statements will make the code look not that elegant. Assuming that this component will be reworked and the preloader will be removed in advance of showing the file component with progress, I suggest leaving as is. Overall, let's strive to avoid this kind of complex notation.